### ISSUE TITLE
Cannot set Firefox preferences when using custom profile

### ISSUE THREAD

**Issue by boolean5** [mozilla/geckodriver#1844]
## System

* Version: 0.29.0
* Platform: Ubuntu 18.04.5 LTS
* Firefox: 87.0a1
* Selenium: 3.141.0 (Python)

## Testcase

```python
import tempfile

from selenium import webdriver
from selenium.webdriver.firefox.options import Options

profile = tempfile.mkdtemp("firefox-profile")

options = Options()
options.binary = "/home/georgia/Downloads/firefox/firefox"
options.log.level = "trace"
options.add_argument("-profile")
options.add_argument(profile)

options.set_preference("browser.startup.page", "https://example.com")
options.set_preference("browser.startup.homepage", "https://example.com")
options.set_preference("browser.startup.firstrunSkipsHomepage", "false")

driver = webdriver.Firefox(options=options, service_args=["--marionette-port", "2828"])
```

### Expected behavior
Firefox should start and load `https://example.com`.

When I don't set a custom Firefox profile through the arguments of `Options` the expected behavior is observed.

### Observed behavior
Firefox starts and loads `https://www.mozilla.org/en-US/firefox/nightly/firstrun/`.

Even though the log shows that Firefox is using the custom profile directory that is specified in the code sample above, geckodriver also creates another profile directory under `/tmp` whose name starts with the `rust_mozprofile` prefix. The only file in that directory is `user.js`, which among other preferences contains the ones set in the code example. As a result, these are not seen by Firefox.

### Notes

https://github.com/mozilla/geckodriver/issues/1058 might also be related to this, as the Marionette port is specified in `user.js`.

## Trace-level log

```
1613485501247   geckodriver     INFO    Listening on 127.0.0.1:58929
1613485502254   mozrunner::runner       INFO    Running command: "/home/georgia/Downloads/firefox/firefox" "--marionette" "-profile" "/tmp/tmpbid8vqz8firefox-profile" "-foreground" "-no-remote"
1613485502255   geckodriver::marionette DEBUG   Waiting 60s to connect to browser on 127.0.0.1:2828
1613485507410   geckodriver::marionette DEBUG   Connection to Marionette established on 127.0.0.1:2828.
1613485507487   webdriver::server       DEBUG   <- 200 OK {"value":{"sessionId":"9f98ab93-8f2f-450d-81a2-548fa3377ab3","capabilities":{"acceptInsecureCerts":true,"browserName":"firefox","browserVersion":"87.0a1","moz:accessibilityChecks":false,"moz:buildID":"20210215162656","moz:geckodriverVersion":"0.29.0","moz:headless":false,"moz:processID":19027,"moz:profile":"/tmp/tmpbid8vqz8firefox-profile","moz:shutdownTimeout":60000,"moz:useNonSpecCompliantPointerOrigin":false,"moz:webdriverClick":true,"pageLoadStrategy":"normal","platformName":"linux","platformVersion":"4.15.0-135-generic","rotatable":false,"setWindowRect":true,"strictFileInteractability":false,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"unhandledPromptBehavior":"dismiss and notify"}}}
```

**Comment by whimboo** [comment]
@jgraham I assume this is just what you are working on via https://bugzilla.mozilla.org/show_bug.cgi?id=1686707?

**Comment by gunesacar** [comment]
> @boolean5 does that work better now with geckodriver 0.30.0 released?

This seems to be fixed in 0.30.0. Using @boolean5's code and geckodriver 0.30.0 I can see that example.com is loaded on the homepage.

**Comment by whimboo** [comment]
Thanks for the verification @gunesacar! It also reminded me to actually close this issue now that it is fixed with the 0.30.0 release.

**Comment by boolean5** [comment]
@whimboo, sorry for getting to this so late and thank you for fixing this issue! @gunesacar, thanks for confirming it works!
